include/circt/Dialect/Seq/Seq.td

```diff
@@ -52,4 +52,46 @@ def CompRegOp : SeqOp<"compreg",
   let hasCustomAssemblyFormat = 1;
 }
 
+def FirRegOp : SeqOp<"firreg",
+    [NoSideEffect, AllTypesMatch<["input", "data"/*, "resetValue"*/]>,
+     SameVariadicOperandSize,
+     DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>]> {
+
+  let summary = "Register with preset and sync or async reset";
+  let description = [{
+    See the Seq dialect rationale for a longer description.
+
+    ```
+    %name = seq.firreg %input, %clk, @sym
+        [ reset (sync|async) %reset, %value ]
+        [ randomised ] : type
+    ```
+  }];
+
+  let arguments = (ins AnyType:$input, I1:$clk, StrAttr:$name,
+                       SymbolNameAttr:$inner_sym,
+                       Optional<I1>:$reset, Optional<AnyType>:$resetValue,
+                       BoolAttr:$isAsync, UnitAttr:$isRandomised);
```

Comment:
I know this was discussed in the rationale document, but now that I look at the PR, I have another question about this unit attr. Is this ever not set? It seems to me like it's always set, and thinking back on how FIRRTL registers work, that makes sense, since FIRRTL relies on randomization to avoid defining things related to X. Am I missing something, or will the unit attr always be set? If it's always set, I suggest just removing it... if there isn't a use-case in the short term, let's not try to predict the future. It's less moving parts for now, and it's simple to add it with a default later if necessary.

Reply:
@seldridge talked about attributes controlling constant preset values for FPGAs. At some point, that will become another optional attribute.

Reply:
Thanks!